docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1015 @@
+
+### 3.1. Ui Component 
+(Jia Ern)
+
+<p align="center">
+  <img src="DG_Images/ui_component.PNG" width="600" alt="Class Diagram of Ui Component"/>
+  <br/>Figure <>. Class diagram of Ui component 
+</p>
+
+The Ui component consists of a main class — `Ui.java`. While `Kaji` has `Ui` as its class variable so that it can instantiate a `Ui` object at each run of the program, the `Model`, `Logic` and `Storage` components have a dependency on the Ui component due to the need to take in user input and show the results of execution.
+
+The Ui component is responsible for:
+* Taking in user input.
+* Printing result of execution for the different commands. 
+* Printing error messages when certain exceptions occur. 
+
+### 3.2. Logic Component 
+(Jane)
+
+The Logic component consists of the `Parser`, `Command` and `Scheduler` classes.
+
+<p align="center">
+  <img src="DG_Images/LogicClassDiagram.png" width="600" alt="Logic Class Diagram"/>
+  <br/>Figure <>. Class diagram of Logic component  
+</p>
+
+1. `Kaji` uses the `Parser` class to parse the user command.
+2. This results in a `Command` object which is executed by `Kaji`.
+3. The command execution can affect the `Model` and `Storage` (e.g. adding a module).
+4. The `Scheduler` is used in some command execution to schedule the due date of a chapter.
+5. The result of the command execution is passed to the `Ui`.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parse("edit 1 CS2113T")` API call:
+
+<p align="center">
+  <img src="DG_Images/LogicSequenceDiagram.png" width="600" alt="Logic Sequence Diagram"/>
+  <br/>Figure <>. Sequence diagram of Logic component  
+</p>
+
+>:information_source: <b>Note:</b> The lifeline for `Parser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+### 3.3. Model Component
+(Jiayi)
+API: Model.java
+
+The Model,
+
+* stores a Access object that holds temporary data for user's access level and contents of level being accessed.
+* stores a History object that saves revision history of current day.
+* stores a DueChapter object that manages due chapter data.
+* contains an Admin class that can be created and accessed by Access object. The Admin class contains a ModuleList object to manage data for modules.
+* contains a Module class that can be created and accessed by Access object. The Module class contains a ChapterList object to manage data for chapters.
+* contains a Chapter class that can be created and accessed by Access object. The Chapter class contains a CardList object to manage data for cards.
+* contains a Card object that can be created and accessed by CardList. A Card object represents a flashcard with question, answer and master level of revision.
+
+<p align="center">
+  <img src="UML/model.png" width="800" alt="Architecture Diagram of Model"/>
+  <br/>Figure <>. Architecture Diagram of Model
+</p>
```

Comment:
To add on to what the others have said, you could consider numbering your diagrams?